### PR TITLE
fix status automatically removed bug

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
The status fields in the workload disappear after a reconcile loop.  The culprit is that the controller runtime merge function doesn't know how to patch a simple Json object... The solution is to manually patch it on the clientside (lame)

### CODE REVIEW

**Review by hongchaodeng — APPROVED** [review]
code lgtm
Might need add some UT to pass codecov.

File: pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go [hunk 1]
```diff
@@ -346,10 +345,36 @@ func (r *OAMApplicationReconciler) updateStatus(ctx context.Context, ac *v1alpha
 		}
 	}
 	ac.Status.Workloads = append(ac.Status.Workloads, revisionStatus...)
-
+	// patch the extra fields in the status that is wiped by the Status() function
```
Comment (hongchaodeng) [inline pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go:348]: `Status()` is not specific enough to pinpoint the code location.
Better point out its parent function/object as well.

File: pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go [hunk 2]
```diff
@@ -346,10 +345,36 @@ func (r *OAMApplicationReconciler) updateStatus(ctx context.Context, ac *v1alpha
...
+	patchExtraStatusField(&ac.Status, acPatch.Status)
 	ac.SetConditions(v1alpha1.ReconcileSuccess())
 }
 
+func patchExtraStatusField(acStatus *v1alpha2.ApplicationConfigurationStatus, acPatchStatus v1alpha2.ApplicationConfigurationStatus) {
```
Comment (wonderflow) [inline pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go:353]: This function has  a problem, workload status which should be removed will always leave here now.

**Review by wonderflow — APPROVED** [review]
LGTM

**Comment by wonderflow** [comment]
The Coverage seems to have some problem, merging

**Pull request merged**